src/operators.rs

```diff
@@ -508,41 +508,360 @@ impl StatefulBatchLogic {
 
         Ok((emit, is_complete))
     }
+}
+
+pub(crate) struct StatefulBatchState {
```

Comment:
This was in the recovery module initially, I forgot to remove the `pub(crate)` after moving it. Will make this private

Reply:
Nope, this (and the other states) are `pub(crate)` in the crate because they are initialized in the worker module. I made all the methods other than `init` private though. We could move the initialization into the state itself though, but I left it as it is for now.